### PR TITLE
Dev

### LINKED ISSUE #3: Workflows Not Running
This issue is regarding one of the github actions not running properly.

### PR DESCRIPTION
**Pull request by NabilFayak**
### Pull Request Description

- Added in relevant workflows (`lint_check`, `pull_request_check`, `release_notes_check`)
- Updated workflows to use repo specific token
- Refactored documentation to fit `CheckMate`
- PDM should be ready for deployment

Resolves #3 
-----
*After creating the pull request: in order to pass the **release_notes_updated** check you will need to update the "Future Release" section of* `docs/source/release_notes.rst` *to include this pull request by adding :pr:`123`.*

### CODE REVIEW

**Comment by machineFL** [comment]
No linked issues found. Please add the corresponding issues in the pull request description. <br/> [Use GitHub automation to close the issue when a PR is merged](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)

**Review by bchen1116 — COMMENTED** [review]
Can we also add in the tests themselves so we can ensure they're passing on the branch?

File: docs/source/_templates/layout.html [hunk 1]
```diff
@@ -9,23 +9,23 @@
...
 </script> -->
 
-<title>EvalML - Alteryx Open Source AutoML Library</title>
-<meta name="description" content="EvalML is an AutoML library that builds, optimizes, and evaluates machine learning pipelines using domain-specific objective functions. — EvalML documentation">
+<title>CheckMate - Alteryx Open Source AutoML Library</title>
+<meta name="description" content="CheckMate is an AutoML library which catches and warns of problems with your data and problem setup before modeling. - CheckMate documentation">
 
 <!-- Facebook Meta Tags -->
-<meta property="og:url" content="https://evalml.alteryx.com/en/stable/">
+<meta property="og:url" content="https://checkmate.alteryx.com/en/stable/">
```
Comment (bchen1116) [inline docs/source/_templates/layout.html]: I don't think there's anything at this site? We can comment this out for now

File: docs/source/_templates/layout.html [hunk 2]
```diff
@@ -9,23 +9,23 @@
...
 
 <!-- Twitter Meta Tags -->
 <meta name="twitter:card" content="summary_large_image">
-<meta property="twitter:domain" content="evalml.alteryx.com">
-<meta property="twitter:url" content="https://evalml.alteryx.com/en/stable/">
-<meta name="twitter:title" content="EvalML - Alteryx Open Source AutoML Library">
-<meta name="twitter:description" content="EvalML is an AutoML library that builds, optimizes, and evaluates machine learning pipelines using domain-specific objective functions. — EvalML documentation">
-<meta name="twitter:image" content="https://featurelabs-static.s3.amazonaws.com/OpenSource_OpenGraph_1200x630px-evalml.png">
+<meta property="twitter:domain" content="checkmate.alteryx.com">
+<meta property="twitter:url" content="https://checkmate.alteryx.com/en/stable/">
```
Comment (bchen1116) [inline docs/source/_templates/layout.html:25]: same here

File: docs/source/release_notes.rst [hunk 3]
```diff
@@ -2,28 +2,32 @@ Release Notes
...
     * Transfer over base `Data Checks` and `IDColumnData Checks` from the `EvalML` repo :pr:`1`
+    * Added in github workflows that are relevant to `DataChecks`, from `EvalML` repository, and modified to fit `DataChecks` wherever possible :pr:`1`
+    * Implemented linters and have them successfully running :pr:`1`
 * Fixes
     * Fixed pytest failures :pr:`1`
+    * Workflows are now up and running properly :pr:`1`
 * Changes
     * Removed all `EvalML` dependencies and unnecessary functions/comments from `utils`, `tests`, `exceptions`, and `datachecks` :pr:`1`
+    * Updated comments to reflect `DataChecks` repository :pr:`1`
+    * Restructured file directory to categorize data checks between `datacheck_meta` and `checks` :pr:`1`
```
Comment (bchen1116) [inline docs/source/release_notes.rst:28]: These numbers should all be `:pr:2` right?

File: docs/source/_templates/layout.html (hunk 1 above)
Comment (NabilFayak) [reply docs/source/_templates/layout.html]: commented out

File: docs/source/_templates/layout.html (hunk 2 above)
Comment (NabilFayak) [reply docs/source/_templates/layout.html:25]: commented out

File: docs/source/release_notes.rst (hunk 3 above)
Comment (NabilFayak) [reply docs/source/release_notes.rst:28]: these were all pushed within the first pr for this repo, when the original repository was shifted to this

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
+---
+name: Feature Request
+about: Suggest an idea for this project
+title: ''
+labels: 'new feature'
+assignees: ''
+
+---
+
+- As a [user/developer], I wish I could use DataChecks to ...
```
Comment (ParthivNaresh) [inline .github/ISSUE_TEMPLATE/feature_request.md]: `DataChecks` or `CheckMate`?

File: checkmate/__init__.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
+"""General datachecks directory."""
```
Comment (ParthivNaresh) [inline checkmate/__init__.py]: Do we want to use `checkmate` here?

File: docs/source/_templates/layout.html [hunk 6]
```diff
@@ -9,23 +9,23 @@
   }}();
 </script> -->
 
-<title>EvalML - Alteryx Open Source AutoML Library</title>
-<meta name="description" content="EvalML is an AutoML library that builds, optimizes, and evaluates machine learning pipelines using domain-specific objective functions. — EvalML documentation">
+<title>CheckMate - Alteryx Open Source AutoML Library</title>
```
Comment (ParthivNaresh) [inline docs/source/_templates/layout.html]: `Alteryx Open Source Data Checks for Machine Learning Library` or something similar

File: docs/source/_templates/layout.html [hunk 7]
```diff
@@ -9,23 +9,23 @@
...
+<meta name="description" content="CheckMate is an AutoML library which catches and warns of problems with your data and problem setup before modeling. - CheckMate documentation">
```
Comment (ParthivNaresh) [inline docs/source/_templates/layout.html]: Not an `AutoML` library, maybe change to something like above

File: docs/source/start.ipynb [hunk 8]
```diff
@@ -11,16 +11,7 @@
...
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": null,
-   "metadata": {},
-   "outputs": [],
-   "source": [
-    ""
+    "In this guide, we'll show how you can use CheckMate to automatically catch and warn of problems with your data and problem setup before modeling.. Along the way, we'll highlight CheckMate's built-in tools and features for understanding and interacting with the process."
```
Comment (ParthivNaresh) [inline docs/source/start.ipynb]: `before modeling.`

**Review by ParthivNaresh — APPROVED** [review]
Looks good! Just some small changes, also I'd recommend using a branch name and PR title that makes it easy to recognize what's going on, especially if we ever want to look back on this to see what changes happened and when

**Comment by NabilFayak** [comment]
> Can we also add in the tests themselves so we can ensure they're passing on the branch?

Just added in the testing workflow

File: checkmate/__init__.py (hunk 5 above)
Comment (NabilFayak) [reply checkmate/__init__.py]: switched

File: .github/ISSUE_TEMPLATE/feature_request.md (hunk 4 above)
Comment (NabilFayak) [reply .github/ISSUE_TEMPLATE/feature_request.md]: Switched to CheckMate

File: docs/source/_templates/layout.html (hunk 6 above)
Comment (NabilFayak) [reply docs/source/_templates/layout.html]: refactored

File: docs/source/_templates/layout.html (hunk 7 above)
Comment (NabilFayak) [reply docs/source/_templates/layout.html]: refactored to fit open source library

File: docs/source/start.ipynb (hunk 8 above)
Comment (NabilFayak) [reply docs/source/start.ipynb]: fixed

File: .github/workflows/pytest.yaml [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+          fetch-depth: 2
+      - name: Installing Dependencies
+        run: |
+          pip install virtualenv
+          virtualenv test_python -q
+          source test_python/bin/activate
+          make installdeps
+          make installdeps-test
+          pip freeze
+          deactivate
```
Comment (bchen1116) [inline .github/workflows/pytest.yaml:34]: Why are we deactivating here?

Comment (NabilFayak) [reply .github/workflows/pytest.yaml:34]: didnt work until i did so

Comment (NabilFayak) [reply .github/workflows/pytest.yaml:34]: so deactivated the venv and reactivated and it works

**Review by bchen1116 — APPROVED** [review]
Left a question but otherwise LGTM

Agreed with Parthiv's comment. In the future, let's try to make the titles and descriptions more informative, so we know what's going on in each issue.

File: .github/workflows/pytest.yaml (hunk 9 above)
Comment (bchen1116) [reply .github/workflows/pytest.yaml:34]: Interesting, ok

**Pull request merged**